src/libs/PersonalDetailsUtils.ts

```diff
@@ -24,10 +25,28 @@ Onyx.connect({
     },
 });
 
+/**
+ * Index for the substring method to remove the merged account prefix.
+ */
+const substringStartIndex = CONST.MERGED_ACCOUNT_PREFIX.length;
+
 function getDisplayNameOrDefault(passedPersonalDetails?: Partial<PersonalDetails> | null, defaultValue = '', shouldFallbackToHidden = true): string {
-    const displayName = passedPersonalDetails?.displayName ? passedPersonalDetails.displayName.replace(CONST.REGEX.MERGED_ACCOUNT_PREFIX, '') : '';
-    const fallbackValue = shouldFallbackToHidden ? Localize.translateLocal('common.hidden') : '';
-    return displayName || defaultValue || fallbackValue;
+    let displayName = passedPersonalDetails?.displayName ?? '';
+    /**
+     * If the displayName starts with the merged account prefix, remove it.
+     */
+    if (displayName.startsWith(CONST.MERGED_ACCOUNT_PREFIX)) {
```

Comment:
Is this really much faster than using `passedPersonalDetails.displayName.replace(CONST.REGEX.MERGED_ACCOUNT_PREFIX, '')`?

I heard recently from @tomekzaw that regexes are known to be much slower in Hermes than in other engines. maybe we can try to address that at the source? I'd love to not have to tip-toe around using super basic regexes in our JS code...

Reply:
> However, using the applied solution startsWith only appears to be slightly faster, so it's a matter of discussion whether we should use it or not, since it's only slightly faster

benchmarks would make it easier to discuss this. How much is "slightly faster"?

> I very much like this approach but it's not straight forward to fix performance issues in Hermes. We can take it as an action item from this PR and then follow up on that and also remove the startsWith from this PR. Wdyt?

Sure. I'm sure it would be difficult to improve regex performance in Hermes, but let's look into it. He might hate me for suggesting this, but it seems like it could be a good task for @fabioh8010, since he's worked on regexes in Hermes in the past.

Reply:
Hi @roryabraham @hurali97 It looks like an interesting challenge 😅 And definitely more difficult than the indices flag one.

I would suggest creating a follow-up GH issue so we can take a look at it in the future, when I have more capacity to do so. Also, would be great if we could put all evidence we have that Regexes in Hermes are very slow compared to other engines.

Reply:
@fabioh8010 Thank you 👍 

I will create an issue shortly and update it all of the relevant information. Just one thing, I will still keep the following check in the codebase because even if we fix the RegEx performance, we are still trying to replace a substring in a string which doesn't contain it, hence the check for `startsWith` is in place.

```ts
    // If the displayName starts with the merged account prefix, remove it.
    if (displayName.startsWith(CONST.MERGED_ACCOUNT_PREFIX)) {
        displayName = displayName.substring(substringStartIndex);
    }
```

We should only be using replace on a string which contains that substring.